src/PermissionManager.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\og;
+
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Manager for OG permissions.
+ *
+ * @todo Provide an interface.
+ */
+class PermissionManager {
+
+  /**
+   * The OG group manager.
+   *
+   * @var \Drupal\og\GroupManager
+   */
+  protected $groupManager;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The service providing information about bundles.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface
+   */
+  protected $entityTypeBundleInfo;
+
+  /**
+   * Constructs a PermissionManager object.
+   *
+   * @param \Drupal\og\GroupManager $group_manager
+   *   The OG group manager.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The service providing information about bundles.
+   */
+  public function __construct(GroupManager $group_manager, EntityTypeManagerInterface $entity_type_manager, EntityTypeBundleInfoInterface $entity_type_bundle_info) {
+    $this->groupManager = $group_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityTypeBundleInfo = $entity_type_bundle_info;
+  }
+
+  /**
+   * Generates the OG permission list for the given group type.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID of the group for which to generate the permissions.
+   * @param string $bundle_id
+   *   The bundle ID of the group for which to generate the permissions.
+   *
+   * @return array
+   *   The list of permissions.
+   */
+  public function generatePermissionList($entity_type_id, $bundle_id) {
```

Comment:
minor: `getPermissionList`? Generate sounds like it's actually going to save something.